ReactAndroid/src/main/java/com/facebook/react/views/text/ReactBaseTextShadowNode.java

```diff
@@ -483,6 +491,16 @@ public void setFontWeight(@Nullable String fontWeightString) {
     }
   }
 
+  @ReactProp(name = ViewProps.FONT_VARIANT)
+  public void setFontVariant(ReadableArray fontVariantArray) {
+    String fontFeatureSettings = ReactTypefaceUtils.parseFontVariant(fontVariantArray);
+
+    if (!Objects.equals(fontFeatureSettings, mFontFeatureSettings)) {
```

Comment:
I'm not familiar with `Object.equals`, does it compare all values for an array?

Reply:
This isn't comparing the `ReadableArray`, but the `String` returned from `ReactTypefaceUtils.parseFontVariant()` call with `this.mFontFeatureSettings`.

https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#equals(java.lang.Object,%20java.lang.Object)

TBH, I stumbled upon this via a Stackoverflow post when looking for a shorter way to write this check so it was new to me as well ;)

Reply:
I might be wrong here, but looks like `java.util.Objects` and it's static methods was introduced in API level 19 (https://developer.android.com/reference/java/util/Objects). So I think it can possibly crash on Android 4 devices with `NoClassDefFoundError: java.util.Objects`, when `Object.equals` is called. We're investigating the similar crash in react-native-camera here: https://github.com/react-native-community/react-native-camera/issues/2528.

Reply:
I can replace this with https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/util/Objects.java#L57 if required